tldw_Server_API/app/core/Research/discovery/service.py

```
@@ -0,0 +1,699 @@
+
+def _sanitize_warning_metadata(metadata: Mapping[str, Any]) -> dict[str, Any]:
+    cleaned: dict[str, Any] = {}
+    for key, value in metadata.items():
+        cleaned[str(key)] = _sanitize_warning_value(
+            value,
+            sanitize_text=_is_diagnostic_metadata_key(key),
+        )
+    return cleaned
+
+
+def _sanitize_warning_value(value: Any, *, sanitize_text: bool) -> Any:
```

Ruff (0.15.17)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)